### PR TITLE
Add colony name style switch for patreons

### LINKED ISSUE #8377: "Hometown" Achievement Description inaccurate
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.
I am also running the latest (possible) versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies/Structurize [Wiki](https://wiki.minecolonies.ldtteam.com/) or I asked on [discord](https://discord.minecolonies.com/). Or I was sent from discord to open an issue here.

### Minecraft Version

1.18

### MineColonies Version

minecolonies-1.18.1-1.0.731-ALPHA

### Structurize Version

structurize-1.18.1-1.0.401-RELEASE

### Related Mods and their Versions

_No response_

### Current Behavior

image

This description is inaccurate. Now, you only need two adult colonists and an empty bed anywhere.

### Expected Behavior

Revise the text for the achievement.

### Reproduction Steps

Open the achievements menu (keybind: L by default).
Mouse over "Hometown" 

### Logs

N/A

### Anything else?

_No response_

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### LINKED ISSUE #8571: [BUG] Rack outline does not appear through walls when searching inventory of a building
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.19

### MineColonies Version

1.19.1-1.0.921-ALPHA

### Structurize Version

1.19.1-1.1.423-ALPHA

### Related Mods and their Versions

Forge: 43.1.1
BlockUI: 1.19-0.0.60-ALPHA
Domum: 1.19-1.0.59-ALPHA
Multi-Piston: 1.19-1.2.16-ALPHA

### Current Behavior

When hitting the ? next to the item in the inventory of the main building block (the chest) the outline shows up around the correct rack but does not show through walls. Neither red no green outlines do this

### Expected Behavior

According to Moobien on the Discord, it should show through walls by default

### Reproduction Steps

Start a colony
Place a building
Place an item in a rack(s)
Search for the item
Search for the rack(s)

### Logs

https://gist.github.com/DarthLeahcim/a44eaaf6810f0ad20a161c49aabf11b9

### Anything else?

First noticed on MineColonies v1.19.1-1.0.927-ALPHA with Forge v42.0.9 but could've been an issue before this.
The log may be a bit messy from me accidentally skipping a step or two when quickly creating a town in Creative

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #8377 
Closes #8571
Closes #

# Changes proposed in this pull request:
- Adds some name styles and add a way for users to switch between them (patreons only)
- Remove the config options
- Move it fully to datapacks
- Make the style pack switcher react more quickly

Review please

### CODE REVIEW

**Review by someaddons — DISMISSED** [review]

**Comment by uecasm** [comment]
> Remove the config options

Removing an existing feature for non-patreons is going to annoy people.  Especially for something like this, where the patreon options are never going to be able to cover all possibilities -- imagine for example a streamer who wants to make their citizens get names from their chat members, or a modpack author who wants fantasy names that fit the theme of their pack.

I think there would probably still be sufficient value in having some standard patreon-only name sets "out of the box" without actually removing the config options.

**Comment by MotionlessTrain** [comment]
^ Or when someone who is not a patreon decided to make a new set of names, for other people to use. This happened the past week, someone worked on a Russian name set. It would be quite annoying for them that after assembling, they can't use their own names

**Comment by Talyda** [comment]
Just chiming in really to agree with both the above!

File: src/main/java/com/minecolonies/coremod/client/render/worldevent/HighlightManager.java [hunk 1]
```diff
@@ -42,13 +43,14 @@ static void render(final WorldEventContext ctx)
...
-                    WorldRenderMacros.GLINT_LINES_WITH_WIDTH), ctx.poseStack, boxRenderData.pos, boxRenderData.argbColor, 0.01f);
-
+                WorldRenderMacros.renderLineBox(Minecraft.getInstance().renderBuffers().bufferSource().getBuffer(
+                    WorldRenderMacros.GLINT_LINES_WITH_WIDTH), ctx.poseStack, boxRenderData.pos, boxRenderData.argbColor, 0.025f);
                 if (!boxRenderData.text.isEmpty())
                 {
-                    WorldRenderMacros.renderDebugText(boxRenderData.pos, boxRenderData.text, ctx.poseStack, true, 3, ctx.bufferSource);
+                   WorldRenderMacros.renderDebugText(boxRenderData.pos, boxRenderData.text, ctx.poseStack, true, 3, ctx.bufferSource);
                 }
+                Minecraft.getInstance().renderBuffers().bufferSource().endBatch();
```
Comment (Nightenom) [inline src/main/java/com/minecolonies/coremod/client/render/worldevent/HighlightManager.java]: Nonono

**Comment by Raycoms** [comment]
You of all people don't see that people can still datapack override the default language? XD

Edit, this was for @uecasm =P

-------- Original Message --------

**Review by MotionlessTrain — DISMISSED** [review]
At least the names part, after the explanation on discord. I'm not sure at all about the rendering code 😄

**Review by someaddons — APPROVED** [review]

**Pull request merged**